Deprecate actor materializer (#27538)
akka/akka: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md

* `ActorSource.actorRef` relying on `PartialFunction` has been replaced in the Java API with a variant more suitable to be called by Java.
* `toUntyped` has been renamed to `toClassic`.

## Akka Stream changes

### Materializer changes

A default materializer is now provided out of the box. For the Java API just pass `system` when running streams,
for Scala an implicit materializer is provided if there is an implicit `ActorSystem` available. This avoids leaking
materializers and simplifies most stream use cases somewhat.

The `ActorMaterializer` factories has been deprecated and replaced with a few corresponding factories in `akka.stream.Materializer`.

> New factories with per-materializer settings has not been provided but should instead be done globally through config or per stream, 

Reviewer: ```suggestion
New factories with per-materializer settings have not been provided. Configuration should instead be done globally through config or per stream using attributes, 
```

see below for more details. 

Having a default materializer available means that most, if not all, usages of Java `ActorMaterializer.create()` 
and Scala `implicit val materializer = ActorMaterializer()` should be removed. 

Details about the stream materializer can be found in [Actor Materializer Lifecycle](../stream/stream-flows-and-basics.md#actor-materializer-lifecycle)

> When using streams from typed the same factories and methods for creating materializers and running streams as from classic can now be used with typed. The  

Reviewer: ```suggestion
When using streams from typed, the same methods can be used for running streams as from classic can now be used. The  
```

Or are there any reasons to still use factories?
